### PR TITLE
[consensus] support committee with 1 node

### PR DESCRIPTION
**Pull request by mwtian**
## Description 

Obviously there is no fault tolerance. But this should be useful for running local cluster cheaply. `min_round_delay` should be set to a higher value for local cluster so not too many consensus blocks are built.

## Test plan 

Add unit and simtest.

---

## Release notes

Check each box that your changes affect. If none of the boxes relate to your changes, release notes aren't required.

For each box you select, include information after the relevant heading that describes the impact of your changes that a user might notice and any actions they must take to implement updates. 

- [ ] Protocol: 
- [ ] Nodes (Validators and Full nodes): 
- [ ] Indexer: 
- [ ] JSON-RPC: 
- [ ] GraphQL: 
- [ ] CLI: 
- [ ] Rust SDK:
- [ ] REST API:

### CODE REVIEW

**Review by arun-koshy — APPROVED** [review]
nice!

**Comment by stefan-mysten** [comment]
You rock @mwtian, thanks a lot!

**Comment by stefan-mysten** [comment]
As I am a bit of a noob in this area, does this change restrict the committee size to only 1?

**Comment by mwtian** [comment]
No problem! This change just makes sure 1-validator committee works in the scenarios covered by the tests (with epoch change). Using any other number (2,3,4,...) of validators in a committee is still supported.

File: consensus/core/src/synchronizer.rs [hunk 1]
```diff
@@ -753,12 +754,17 @@ impl<C: NetworkClient, V: BlockVerifier, D: CoreThreadDispatcher> Synchronizer<C
...
                 let mut retry_delay_step = Duration::from_millis(500);
                 'main:loop {
-                    total_stake = 0;
+                    if context.committee.size() == 1 {
+                        highest_round = dag_state.read().get_last_proposed_block().round();
+                        info!("Only one node in the network, will not try fetching own last block from peers.");
+                        break 'main;
+                    }
+
+                    let mut total_stake = 0;
```
Comment (akichidis) [inline consensus/core/src/synchronizer.rs:767]: Nice fix as well moving the `total_stake` here.

**Review by akichidis — APPROVED** [review]
Nice one @mwtian 👍 !

**Comment by stefan-mysten** [comment]
Thanks a lot @mwtian!!! 🚀

**Pull request merged**